core/dbt/parser/manifest.py

```diff
@@ -1385,7 +1546,20 @@ def _process_refs_for_node(manifest: Manifest, current_project: str, node: Manif
 
         target_model_id = target_model.unique_id
 
-        node.depends_on.add_node(target_model_id)
+        if target_model.is_public_node:
+            node.depends_on.add_public_node(target_model_id)
+        else:
+            node.depends_on.add_node(target_model_id)
+
+
+def remove_dependent_project_references(manifest: Manifest, publication: PublicationConfig):
+    for unique_id in publication.public_model_ids:
+        for child_id in manifest.child_map[unique_id]:
+            node = manifest.expect(child_id)
+            if hasattr(node.depends_on, "public_nodes"):
```

Comment:
not blocking, for my own understanding - which nodes dont have depends_on.public_nodes?

Reply:
SeedNodes, Macro, SourceDefinition